packages/webdriverio/src/commands/browser/newWindow.ts

```diff
@@ -3,59 +3,97 @@ import { sleep } from '@wdio/utils'
         await browser.switchToWindow(handles[0])
         console.log(await browser.getTitle()) // outputs: "Google"
     });
  * </example>
+ * <example>
+      :newTabSync.js
+      it('should open a new tab', async () => {
+          await browser.url('https://google.com')
+            console.log(await browser.getTitle()) // outputs: "Google"
+
+         await browser.newWindow('https://webdriver.io', {
+              type:'tab',
+              windowName: 'WebdriverIO window',
+              windowFeature: 'width=420,height=230,resizable,scrollbars=yes,status=1',
+          })
+          console.log(await browser.getTitle()) // outputs: "WebdriverIO · Next-gen browser and mobile automation test framework for Node.js"
+          console.log(result.type) // outputs: "tab"
+          const handles = await browser.getWindowHandles()
+          await browser.switchToWindow(handles[1])
+          await browser.closeWindow()
+          await browser.switchToWindow(handles[0])
+          console.log(await browser.getTitle()) // outputs: "Google"
+     });
+ * </example>
  *
  * @param {string}  url      website URL to open
  * @param {NewWindowOptions=} options                newWindow command options
+ * @param {string=}           options.type         type of new window: 'tab' or 'window'
  * @param {String=}           options.windowName     name of the new window
  * @param {String=}           options.windowFeatures features of opened window (e.g. size, position, scrollbars, etc.)
  *
- * @return {String}          id of window handle of new tab
+ * @return {Object}          An object containing the window handle and the type of new window
+ * @return {String}          handle - The ID of the window handle of the new tab or window
+ * @return {String}          type - The type of the new window, either 'tab' or 'window'
+ *
+ * @throws {Error} If `url` is invalid, if the command is used on mobile, or `type` is not 'tab' or 'window'.
  *
  * @uses browser/execute, protocol/getWindowHandles, protocol/switchToWindow
  * @alias browser.newWindow
- * @type window
+ * @type window or tab
  */
 export async function newWindow (
     this: WebdriverIO.Browser,
     url: string,
-    { windowName = '', windowFeatures = '' }: NewWindowOptions = {}
-) {
+    { type = 'window', windowName = '', windowFeatures = '' }: NewWindowOptions = {}
+): Promise<{ handle: string, type: 'tab' | 'window' }> {
     /**
      * parameter check
      */
     if (typeof url !== 'string') {
         throw new Error('number or type of arguments don\'t agree with newWindow command')
     }
 
+    /**
+    * Validate the 'type' parameter to ensure it is either 'tab' or 'window'
+    */
+    if (!['tab', 'window'].includes(type)) {
+        throw new Error(`Invalid type '${type}' provided to newWindow command. Use either 'tab' or 'window'`)
+    }
+
+    if (windowFeatures) {
+        log.warn('The "windowFeatures" option is deprecated and will be removed in future versions.')
+    }
```

Comment:
We should just generally deprecate this

```suggestion
    if (windowName || windowFeatures) {
        log.warn('The "windowName" and "windowFeatures" options are deprecated and only supported in WebDriver Classic sessions.')
    }
```